extensions/ql-vscode/src/view/data-extensions-editor/MethodRow.tsx

```diff
@@ -51,6 +52,7 @@ const modelTypeOptions: Array<{ value: ModeledMethodType; label: string }> = [
 type Props = {
   externalApiUsage: ExternalApiUsage;
   modeledMethod: ModeledMethod | undefined;
+  modifiedSignatures: Set<string>;
```

Comment:
Yep. I've moved this component to a separate file. Sorry I have a habit of placing small new components in the same file when I think they won't likely be reused. But I agree with you it's more clear if everything is separated out.

Reply:
I tried this but unfortunately it turned out a little complicated. For one thing the logic of `modifiedSignatures.has(externalApiUsage.signature)` in `MethodRow` doesn't translate perfectly to passing in a `ModelingStatus` value. This logic is a little bit of a hack though so I expect we'll change it in the future. I could add a note to the issue about that that we could also clean this up.

Reply:
I'm not sure I quite understand why it's hard to change this. It seems like within `ModelableMethodRow` and `MethodRow` we're only using `modifiedSignatures.has(externalApiUsage.signature)`. If we change this prop to be `unsavedModification: boolean` and change the `ModeledMethodDataGrid` to use `unsavedModification={modifiedSignatures.has(externalApiUsage.signature)}`, wouldn't that be enough? I don't think we need to pass in the `ModelingStatus` value, just the single boolean is enough.

Reply:
The difficulty is that we also pass it into `ModelableMethodRow` and there we do
```typescript
const modificationState = useMemo(() => {
  if (modeledMethod) {
    if (modifiedSignatures.has(modeledMethod.signature)) {
      return "unsaved";
    } else if (modeledMethod.type !== "none") {
      return "saved";
    }
  }
  return "unmodeled";
}, [modeledMethod, modifiedSignatures]);
```
And this can't be deduced from the value of `modifiedSignatures.has(externalApiUsage.signature)`.

(Side note that I now realise this doesn't necessarily have to be `useMemo` since it's a string value and hence uses equality of the string value rather than object equality)

Reply:
Ah, I see. Isn't it the case though that the `externalApiUsage.signature === modeledMethod.signature` in all cases, so we could replace it by `modifiedSignatures.has(externalApiUsage.signature)`?